src/book/mod.rs

```diff
@@ -262,6 +262,10 @@ impl MDBook {
         let mut index = File::create(&themedir.join("index.hbs"))?;
         index.write_all(theme::INDEX)?;
 
+        // header.hbs
+        let mut header = File::create(&themedir.join("header.hbs"))?;
```

Comment:
and when the "header.hbs" is missing?
I understand that this should be handled similarly to "additional.css/js". Hmm We might be interested in extracting some type hbs partials registry.

Reply:
So this works similar to other /present/ files in the theme, in that there's already a blank `header.hbs` file and when the user doesn't supply one, the blank one is used, so it does nothing. I don't think I'm clear on what the difference is.

If we want to be more free form we could just let the user specify partials and then they can override the `index.hbs` itself to use their own partials in their own places (maybe this is one way to address CSS problems).